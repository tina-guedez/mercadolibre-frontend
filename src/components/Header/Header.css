.nav {
  background-color: #ffe600;
  height: 60px;
  position: relative;
}
.nav-menu {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 20px;
  padding: 0 50px;
}
.nav-logo {
  background-image: url('../../assets/Logo_ML@2x.png.png.png');
  background-repeat: no-repeat;
  background-size: cover;
  grid-column: 2 / 3;
  grid-row: 1;
  height: 40px;
  margin: 10px 0;
  margin-right: 20px;
  width: 59px;
}
