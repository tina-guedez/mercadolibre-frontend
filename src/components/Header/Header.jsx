import React from 'react';
import './Header.css';
import { Link } from 'react-router-dom';
import SearchBar from '../SearchBar/SearchBar';

export default function Header() {
  return (
    <header className="nav">
      <div className="nav-menu">
        <Link to="/" tabIndex="0" className="nav-logo" />
        <SearchBar />
      </div>
    </header>
  );
}
