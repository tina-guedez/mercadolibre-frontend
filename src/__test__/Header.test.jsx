import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../components/Header/Header';

describe('Pruebas en el componente Header', () => {
  test('Verifica que el componente se renderice', () => {
    const { container } = render(<Router><Header /></Router>);
    expect(container.outerHTML).not.toBeUndefined();
  });
  it('Si hace match con el snapshot', () => {
    const { container } = render(<Router><Header /></Router>);
    expect(container).toMatchSnapshot();
  });
});
