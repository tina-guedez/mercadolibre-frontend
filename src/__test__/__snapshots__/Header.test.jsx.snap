// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Header Si hace match con el snapshot 1`] = `
<div>
  <header
    class="nav"
  >
    <div
      class="nav-menu"
    >
      <a
        class="nav-logo"
        href="/"
        tabindex="0"
      />
      <form
        action="/items"
        class="nav-search"
        method="GET"
        role="search"
      >
        <input
          aria-label="Ingresa el producto a buscar"
          autocomplete="off"
          class="nav-search-input"
          maxlength="120"
          name="q"
          placeholder="Nunca dejes de buscar"
          required=""
          tabindex="0"
          type="text"
        />
        <button
          class="nav-search-btn"
          tabindex="-1"
          type="submit"
        >
          <img
            aria-label="Buscar"
            class="nav-icon-search"
            src="test-file-stub"
          />
        </button>
      </form>
    </div>
  </header>
</div>
`;
